extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourcesJdbcBuildTimeConfig.java

```diff
@@ -25,6 +28,13 @@ public interface DataSourcesJdbcBuildTimeConfig {
     @WithUnnamedKey(DataSourceUtil.DEFAULT_DATASOURCE_NAME)
     Map<String, DataSourceJdbcOuterNamedBuildTimeConfig> dataSources();
 
+    /**
+     * Dev UI.
+     */
+    @WithDefaults
```

Comment:
This annotation is used for maps only according to the docs https://smallrye.io/smallrye-config/Main/config/mappings/#withdefaults. Does it have any function here?